### DISCUSSION TITLE
Simplify intersections with additional constraints on connectivity

### DISCUSSION THREAD

**Discussion by mkvasnicka** [luukvdmeer/sfnetworks#206] [Q&A]
Hi!

First of all, many thanks for the package. It is the best choice to simplify road networks I have come across. Nonetheless, there is a problem with the intersection-simplification algorithm (at least for roads): it creates "lifts" in mixmasters (off-grade intersections), i.e., it connects off-level paths that were not directly connected before.

For instance,  see this case:

```{r}
p11 <- st_point(c(-1, 0))
p12 <- st_point(c(0.1, 0))
p13 <- st_point(c(1, 0))
p21 <- st_point(c(0, -1))
p22 <- st_point(c(0, 0.1))
p23 <- st_point(c(0,1))
lines <- st_sfc(st_linestring(c(p11, p12)),
                st_linestring(c(p12, p13)),
                st_linestring(c(p21, p22)),
                st_linestring(c(p22, p23)),
                st_linestring(c(p11, p21)))
```

Points p12 and p22 should not be contracted, as they are off-level.

The problem is caused by group_components() which is too benevolent. Is there any other function that works similarly but tests that the nodes are connected directly?

Many thanks for your help.
Michal

**Comment by agila5** [comment]
Hi @mkvasnicka. Can you share the complete code that you used to run the contraction?
